src/stp/plays/referee_specific/FreeKickThem.cpp

```diff
@@ -2,31 +2,76 @@
 
 FreeKickThem::FreeKickThem() : Play() {
-    startPlayInvariants.clear();
-    startPlayInvariants.emplace_back(std::make_unique<invariant::FreeKickThemGameStateInvariant>());
+    /// Evaluations that have to be true to be considered when changing plays.
+    startPlayEvaluation.clear();  // DONT TOUCH.
+    startPlayEvaluation.emplace_back(eval::FreeKickThemGameState);
 
-    keepPlayInvariants.clear();
-    keepPlayInvariants.emplace_back(std::make_unique<invariant::FreeKickThemGameStateInvariant>());
+    /// Evaluations that have to be true to allow the play to continue, otherwise the play will change. Plays can also end using the shouldEndPlay().
+    keepPlayEvaluation.clear();  // DONT TOUCH.
+    keepPlayEvaluation.emplace_back(eval::FreeKickThemGameState);
 
+    /// Role creation, the names should be unique. The names are used in the stpInfos-map.
     roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{
-        std::make_unique<role::Keeper>("keeper"),       std::make_unique<role::Defender>("defender_0"), std::make_unique<role::Defender>("defender_1"),
-        std::make_unique<role::Defender>("defender_2"), std::make_unique<role::Defender>("blocker_0"),  std::make_unique<role::Defender>("blocker_1"),
-        std::make_unique<role::Defender>("blocker_2"),  std::make_unique<role::Defender>("blocker_3"),  std::make_unique<role::Defender>("blocker_4"),
-        std::make_unique<role::Defender>("blocker_5"),  std::make_unique<role::Formation>("offender")};
+        std::make_unique<role::Keeper>("keeper"),
+        std::make_unique<role::Defender>("defender_0"),
+        std::make_unique<role::Defender>("defender_1"),
+        std::make_unique<role::Defender>("defender_2"),
+        std::make_unique<role::Defender>("blocker_0"),
+        std::make_unique<role::Defender>("blocker_1"),
+        std::make_unique<role::Defender>("blocker_2"),
+        std::make_unique<role::Defender>("blocker_3"),
+        std::make_unique<role::Defender>("blocker_4"),
+        std::make_unique<role::Defender>("blocker_5"),
+        std::make_unique<role::Formation>("offender")
+    };
+
+    initRoles();  // DONT TOUCH.
 }
 
-uint8_t FreeKickThem::score(world::World *world) noexcept { return 100; }
+Dealer::FlagMap FreeKickThem::decideRoleFlags() const noexcept {
+    Dealer::FlagMap flagMap;  // DONT TOUCH.
+
+    /// Flags that have a factor and a weight linked to it, can be given to a role
+    Dealer::DealerFlag closeToOurGoalFlag(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::HIGH_PRIORITY);
+    Dealer::DealerFlag closeToBallFlag(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::HIGH_PRIORITY);
+
+    /// Creation flagMap. Linking roles to role-priority and the above created flags, can also force ID {roleName, {priority, flags, forceID}}
+    flagMap.insert({"keeper", {DealerFlagPriority::KEEPER, {}}});
+    flagMap.insert({"defender_0", {DealerFlagPriority::HIGH_PRIORITY, {closeToOurGoalFlag}}});
```

Comment:
The DealerFlagPriority's are supposed to be used to give the Dealer an indication of which Roles are most important to a Play. By using them the Dealer can decide the best division of Roles over robots, but if you give each role 'HIGH_PRIORITY' the Dealer won't divide the Roles in an efficient way. My suggestion: Look at the different Roles in this Play, decide which ones are absolutely necessary and give these the 'REQUIRED' priority. Then look at which Roles are most important after that, and give those the 'HIGH_PRIORITY'. Then after that the Roles which are 'MEDIUM_PRIORITY', etc.